Documentation/Development.md

```diff
@@ -6,27 +6,232 @@
+set TF_NEED_CUDA=0
+set TF_CUDA_COMPUTE_CAPABILITIES=7.5
+set CC_OPT_FLAGS="/arch:AVX /D_USE_MATH_DEFINES"
+set TF_OVERRIDE_EIGEN_STRONG_INLINE=1
+.\tensorflow\configure.py
+:: build
+set BAZEL_SH=%ProgramFiles%\Git\usr\bin\bash.exe
+set BAZEL_VC=%VCINSTALLDIR%
+bazel --output_user_root %CD%/caches/bazel/tensorflow build -c opt --copt /D_USE_MATH_DEFINES --define framework_shared_object=false --config short_logs --nocheck_visibility //tensorflow:tensorflow //tensorflow:tensorflow_dll_import_lib //tensorflow/compiler/tf2xla/xla_tensor:x10 //tensorflow/compiler/tf2xla/xla_tensor:x10_dll_import_lib
+:: terminate bazel daemon
+bazel --output_user_root %CD%/caches/bazel/tensorflow shutdown
+
+:: package
+set DESTDIR=%CD%\Library\tensorflow-windows-%VSCMD_ARG_TGT_ARCH%\tensorflow-2.4.0
+
+md %DESTDIR\usr\bin
+copy tensorflow\bazel-bin\tensorflow\tensorflow.dll %DESTDIR%\usr\bin\
+copy tensorflow\bazel-bin\tensorflow\compiler\tf2xla\xla_tensor\x10.dll %DESTDIR%\usr\bin\
+
+md %DESTDIR%\usr\lib
+copy tensorflow\bazel-out\%VSCMD_ARG_TGT_ARCH%_windows-opt\bin\tensorflow\tensorflow.lib %DESTDIR%\usr\lib\
+copy tensorflow\bazel-out\%VSCMD_ARG_TGT_ARCH%_windows-opt\bin\tensorflow\compiler\tf2xla\xla_tensor\x10.lib %DESTDIR%\usr\lib\
+
+md %DESTDIR%\usr\include\tensorflow\c
+copy tensorflow\tensorflow\c\c_api.h %DESTDIR%\usr\include\tensorflow\c\
+copy tensorflow\tensorflow\c\c_api_experimental.h %DESTDIR%\usr\include\tensorflow\c\
+copy tensorflow\tensorflow\c\tf_attrtype.h %DESTDIR%\usr\include\tensorflow\c\
+copy tensorflow\tensorflow\c\tf_datatype.h %DESTDIR%\usr\include\tensorflow\c\
+copy tensorflow\tensorflow\c\tf_file_statistics.h %DESTDIR%\usr\include\tensorflow\c\
+copy tensorflow\tensorflow\c\tf_status.h %DESTDIR%\usr\include\tensorflow\c\
+copy tensorflow\tensorflow\c\tf_tensor.h %DESTDIR%\usr\include\tensorflow\c\
+
+md %DESTDIR%\usr\include\tensorflow\c\eager
+cp tensorflow\tensorflow\c\eager\c_api.h %DESTDIR%\usr\include\tensorflow\c\eager\
+
+md %DESTDIR%\usr\include\x10
+copy swift-apis\Sources\x10\swift_bindings\device_wrapper.h %DESTDIR%\usr\include\x10\
+copy swift-apis\Sources\x10\swift_bindings\xla_tensor_tf_ops.h %DESTDIR%\usr\include\x10\
+copy swift-apis\Sources\x10\swift_bindings\xla_tensor_wrapper.h %DESTDIR%\usr\include\x10\
+
+md %DESTDIR%\usr\share
+copy tensorflow\bazel-out\%VSCMD_ARG_TGT_ARCH%_windows-opt\bin\tensorflow\tensorflow_filtered_def_file.def %DESTDIR%\usr\share
+```
+</details>
+
+<details>
+    <summary>macOS/Linux Build Script</summary>
+
+```bash
+# clone swift-apis
+git clone git://github.com/tensorflow/swift-apis
+# checkout tensorflow
+git clone --depth 1 --no-tags git://github.com/tensorflow/tensorflow
+git -C tensorflow checkout refs/tags/v2.4.0
+
+# copy X10
+cp -ar swift-apis/Sources/CX10 tensorflow/swift_bindings
+cp -ar swift-apis/Sources/x10/xla_client tensorflow/tensorflow/compiler/xla/xla_client
+cp -ar swift-apis/Sources/x10/xla_tensor tensorflow/tensorflow/compiler/tf2xla/xla_tensor
```

Comment:
On macOS:
```console
$ cp -ar swift-apis/Sources/CX10 tensorflow/swift_bindings
cp: the -R and -r options may not be specified together.
```

I think `-a` is stronger than and subsumes `-r` on macOS? Not sure about Linux.

Reply:
Yes, `-a` should be sufficient, I can reduce this to `-a`.  Although, Im thinking that I no longer like this approach and would prefer the CMake approach - suggesting people use symbolic links.  I want to try to make it easier for people to be able to iterate on X10.

There really are only two scenarios that make sense to build X10 from source:
1. you are building for a target for which we do not have pre-built X10 releases
2. you are making changes for X10

The first case you have no choice and you must build it.  It is the second case that I want to make more approachable.